ompi/mca/coll/base/coll_base_barrier.c

```diff
@@ -61,13 +61,14 @@ ompi_coll_base_sendrecv_zero(int dest, int stag,
     if (err != MPI_SUCCESS) { line = __LINE__; goto error_handler; }
 
     err = ompi_request_wait_all( 2, reqs, statuses );
-    if( MPI_ERR_IN_STATUS == err ) {
+    if( MPI_ERR_IN_STATUS == err ) { line = __LINE__;
         /* As we use wait_all we will get MPI_ERR_IN_STATUS which is not an error
          * code that we can propagate up the stack. Instead, look for the real
          * error code from the MPI_ERROR in the status.
          */
         int err_index = 0;
-        if( MPI_SUCCESS == statuses[0].MPI_ERROR ) {
+        if( MPI_SUCCESS == statuses[0].MPI_ERROR
```

Comment:
@abouteiller i do not fully understand the logic here.

with this code, i can see two cases that might not produce the desired output.
```
statuses[0].MPI_ERROR == MPI_ERR_PENDING
statuses[1].MPI_ERROR == MPI_SUCCESS
```
returns ```MPI_SUCCESS```

```
statuses[0].MPI_ERROR == MPI_SUCCESS
statuses[1].MPI_ERROR == MPI_ERR_PENDING
```
returns ```MPI_ERR_PENDING```

in these two cases, should the function returns an error code different than ```MPI_SUCCESS``` and ```MPI_ERR_PENDING```


as a side note, is ```MPI_Irecv``` / ```MPI_Isend``` optimal here ?

for example, ```MPI_Irecv``` / ```MPI_Send``` could be a bit faster.